tests/draft4/ref.json

```diff
@@ -296,5 +296,110 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "Location-independent identifier",
+        "schema": {
+            "$ref": "#foo",
+            "definitions": {
+                "A": {
+                    "id": "#foo",
+                    "type": "integer"
+                }
+            }
+        },
+        "tests": [
+            {
+                "data": 1,
+                "description": "match",
+                "valid": true
+            },
+            {
+                "data": "a",
+                "description": "mismatch",
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "Canonical location-independent identifier",
```

Comment:
I'd like to avoid the word "canonical" due to its role in the confusing draft-04 canonical vs inline mess.

I'm not entirely sure it applies to this case even in draft-04 terms, but TBH none of the current spec team really know what the point of that was AFAIK.

Is the point of this test to use a URI (with scheme) rather than a URI reference?  If so, that seems like the thing to note in the description.